### PR TITLE
Raw is now using Memslice

### LINKED ISSUE #1024: Consider using MemSlice in header::Raw
The `Raw` type is how hyper stores the raw bytes from a request or response read off a socket. It currently keeps its implementation private, so we could update it to hold on to a `MemSlice` instead of just a `Vec` or static slice, which could reduce copies.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #1024.

I still need to find where I made a mistake to fix the tests. (And add some tests as well)

### CODE REVIEW

File: src/header/raw.rs [hunk 1]
```diff
@@ -1,229 +1,163 @@
-use std::borrow::Cow;
 use std::fmt;
+use http::buf::{self, MemSlice};
 
 /// A raw header value.
 #[derive(Clone, PartialEq, Eq)]
-pub struct Raw(Lines);
+pub struct Raw(MemSlice);
```
Comment (seanmonstar) [inline src/header/raw.rs]: I was actually thinking more that internally, we'd have `Line` change from `Cow` to another enum:

```rust
enum Line {
    Static(&'static [u8]),
    Owned(Vec<u8>),
    Shared(MemSlice),
}
```

Comment (GuillaumeGomez) [inline src/header/raw.rs]: It makes more sense.

**Comment by GuillaumeGomez** [comment]
Ok, new version! Now I just need to find where I could use the `MemSlice` instead of a `Vec<u8>`.

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 4a806eeb9c515974dab34326d973331b08d63f6d
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 99079e8193f9dad6d493811d1f446be701f00a7f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 03f3b42a771af41ade59ca0cb91274d1c8ac12e5
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f84a5369055702570b2ae3b2d34621d4db0f8ca3
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 03f3b42a771af41ade59ca0cb91274d1c8ac12e5
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f84a5369055702570b2ae3b2d34621d4db0f8ca3
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 31b8cba4094550dbcde39c044a8ce1b08f1b2463
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 03f3b42a771af41ade59ca0cb91274d1c8ac12e5
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f84a5369055702570b2ae3b2d34621d4db0f8ca3
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: ecef9b67e99e0e5d1e8cd346ab4187209296f232
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 03f3b42a771af41ade59ca0cb91274d1c8ac12e5
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f84a5369055702570b2ae3b2d34621d4db0f8ca3
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: b977be3243b6f6d9f506ee1c4eb9d1b1d11246d6
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 03f3b42a771af41ade59ca0cb91274d1c8ac12e5
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f84a5369055702570b2ae3b2d34621d4db0f8ca3
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22d75f7cc6295f9c83f5d3c2577ef2a32dc0c9ec
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GuillaumeGomez** [comment]
I even rewrote commits message.

**Pull request closed without merging**